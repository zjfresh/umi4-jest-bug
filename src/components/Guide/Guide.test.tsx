import { render } from '@testing-library/react';

import Guide from './index';

Object.defineProperty(window, 'matchMedia', {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // Deprecated
    removeListener: jest.fn(), // Deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});

describe('RemoteSelect unit test', () => {
  const Text = 'abc';

  test('渲染冒烟测试', async () => {
    const { container } = render(<Guide name={Text} />);
    expect(container).toMatchSnapshot();
  });
});
