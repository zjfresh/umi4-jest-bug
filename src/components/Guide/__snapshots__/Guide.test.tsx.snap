// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RemoteSelect unit test 渲染冒烟测试 1`] = `
<div>
  <section
    class="ant-layout"
  >
    <div
      class="ant-row"
    >
      <h3
        class="ant-typography title"
      >
        欢迎使用 
        <strong>
          abc
        </strong>
         ！
      </h3>
    </div>
  </section>
</div>
`;
